src/Extension.Tests/StageControlTests.cs

```diff
@@ -0,0 +1,222 @@
+﻿using Extension.StageControl;
+using Extension.Tests.Utilities;
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Extension.Tests
+{
+    public class StageControlTests
+    {
+        private string[] sampleStageContent = new string[]
+            {
+                "contentcell1",
+                "contentcell2",
+                "contentcell3"
+            };
+
+        private Stage GetEmptyStage(int stageId = 1)
+        {
+            return new Stage(stageId, Enumerable.Empty<string>());
+        }
+
+        [Fact]
+        public void stage_with_no_dependency_is_auto_revealed_single_stage_case()
+        {
+            List<int> revealedStageIds = new List<int>();
+
+            var stageController = new StageController();
+            stageController.AddBlankStages(1);
```

Comment:
Why ints instead of objects?

Reply:
It is a stageId. StageIds are ints at the moment. But I could make them into more complex types?

Reply:
There is a chance stageId's might be entirely internal, which means we probably won't need stageIds. The teacher sees stage names (strings) and those are mapped to stage objects directly.

Reply:
IDs are useful for tracking identities in external systems (services, databases) but less so when you already have object references. In fact the ID and the reference can get out of sync, so they can be a bit of a liability when everything is in-memory.

Reply:
makes sense. I've removed ids